File: nodejs/summarize-text/basic-auth.js
const https = require("https");

// basic configuration
const clientId = "[[client_id]]";
const clientSecret = "[[client_secret]]";

// create a base64 encoded api key
const apiKey = Buffer.from(`${clientId}:${clientSecret}`).toString("base64");

(async () => {
  try {
    // use the api key as a basic token
    const authorization = `Basic ${apiKey}`;

    // populate endpoint parameters
    const textBody = "[[text_body]]";
    const userAgent = "Intelligent API Sample Nodejs Code";

    const postData = JSON.stringify({
      text: textBody,
    });

    const options = {
      hostname: "api.intelligent-api.com",
      port: 443,
      path: "/v1/text/summarize",
      method: "POST",
      headers: {
        Authorization: authorization,
        "Content-Type": "application/json",
        "Content-Length": Buffer.byteLength(postData),
        "User-Agent": userAgent,
      },
    };

    const request = https.request(options, (response) => {
      console.log(`Status Code: ${response.statusCode}`);

      response.on("data", (chunk) => {
        console.log("Response Body:", chunk.toString());
      });
    });

    request.on("error", (e) => {
      console.error(`Problem with request: ${e.message}`);
    });

    request.write(postData);
    request.end();
  } catch (error) {
    console.error("Error:", error);
  }
})();

File: ruby/translate/basic-auth.rb
require 'base64'
require 'net/http'
require 'json'

# basic configuration
client_id = "[[client_id]]"
client_secret = "[[client_secret]]"

# create a base64 encoded api key
api_key = Base64.strict_encode64("#{client_id}:#{client_secret}")

# use the api key as a basic token
authorization = "Basic #{api_key}"

# populate endpoint parameters
text = "[[text_body]]"
source_language = "[[source_language]]"
target_language = "[[target_language]]"
user_agent = "Intelligent API Sample Ruby Code"

data = {
  text: text,
  sourceLanguage: source_language,
  targetLanguage: target_language
}

json_data = data.to_json

# invoke the API endpoint
url = URI("https://api.intelligent-api.com/v1/language/translate")

http = Net::HTTP.new(url.host, url.port)
http.use_ssl = (url.scheme == "https")

request = Net::HTTP::Post.new(url)
request["Authorization"] = authorization
request["Content-Type"] = "application/json"
request["User-Agent"] = user_agent
request.body = json_data

response = http.request(request)

puts "Response Code: #{response.code}"
puts "Response Body: #{response.body}"